src/main/java/org/sagebionetworks/bridge/spring/controllers/AssessmentController.java

```diff
@@ -0,0 +1,254 @@
+            throw new UnauthorizedException(SHARED_ASSESSMENTS_ERROR);
+        }
+
+        if ("true".equals(physical) && session.isInRole(ADMIN)) {
+            service.deleteAssessmentPermanently(appId, guid);
+        } else {
+            service.deleteAssessment(appId, guid);
+        }
+        return new StatusMessage("Assessment deleted.");        
+    }
+
+    /* === Methods for working with an identifier, not a GUID === */
+        
+    @GetMapping("/v1/assessments/identifier:{identifier}")
+    public AssessmentDto getLatestAssessment(@PathVariable String identifier) {
+        UserSession session = getAuthenticatedSession(DEVELOPER);
+
+        String appId = session.getStudyIdentifier().getIdentifier();
+        if (SHARED_STUDY_ID_STRING.equals(appId)) {
+            throw new UnauthorizedException(SHARED_ASSESSMENTS_ERROR);
+        }
+
+        Assessment retValue = service.getLatestAssessment(appId, identifier);
+        return AssessmentDto.create(retValue);
+    }
+    
+    @GetMapping("/v1/assessments/identifier:{identifier}/revisions")
+    public PagedResourceList<AssessmentDto> getAssessmentRevisionsById(@PathVariable String identifier,
+            @RequestParam(required = false) String offsetBy, @RequestParam(required = false) String pageSize,
+            @RequestParam(required = false) String includeDeleted) {
+        UserSession session = getAuthenticatedSession(DEVELOPER);
+
+        String appId = session.getStudyIdentifier().getIdentifier();
+        if (SHARED_STUDY_ID_STRING.equals(appId)) {
+            throw new UnauthorizedException(SHARED_ASSESSMENTS_ERROR);
+        }
+
+        int offsetByInt = BridgeUtils.getIntOrDefault(offsetBy, 0);
+        int pageSizeInt = BridgeUtils.getIntOrDefault(pageSize, API_DEFAULT_PAGE_SIZE);
+        boolean incDeletedBool = Boolean.valueOf(includeDeleted);
+        
+        PagedResourceList<Assessment> page = service.getAssessmentRevisionsById(
+                appId, identifier, offsetByInt, pageSizeInt, incDeletedBool);
+
+        List<AssessmentDto> dtos = page.getItems().stream()
+                .map(assessment -> AssessmentDto.create(assessment))
+                .collect(Collectors.toList());
+        
+        return new PagedResourceList<>(dtos, page.getTotal())
+                .withAllRequestParams(page.getRequestParams());
+    }
+    
+    @GetMapping("/v1/assessments/identifier:{identifier}/revisions/{revision}")
+    public AssessmentDto getAssessmentById(@PathVariable String identifier, @PathVariable String revision) {
+        UserSession session = getAuthenticatedSession(DEVELOPER);
+
+        String appId = session.getStudyIdentifier().getIdentifier();
+        if (SHARED_STUDY_ID_STRING.equals(appId)) {
+            throw new UnauthorizedException(SHARED_ASSESSMENTS_ERROR);
+        }
+
+        // 0 is not a valid value, on purpose, will throw BadRequestException
+        int revisionInt = BridgeUtils.getIntOrDefault(revision, 0);
```

Comment:
I'll throw here. Any value can be used in that part of the path because the parameter type is String, and we do that so we can control the error returned and make it similar to internal exceptions. It now throws if it's "0", "-1", "nonsense", etc.